File: baselibrary/src/main/java/com/minicart/android/baselibrary/dagger2/module/AppModule.java
package com.minicart.android.baselibrary.dagger2.module;

import android.app.Application;
import android.support.v4.util.ArrayMap;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.util.Map;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    private Application mApplication;

    public AppModule(Application application) {
        this.mApplication = application;
    }

    @Singleton
    @Provides
    public Application provideApplication() {
        return mApplication;
    }

    @Singleton
    @Provides
    public Gson provideGson(Application application) {
        return new GsonBuilder().create();
    }

    @Singleton
    @Provides
    public Map<String, Object> provideExtras() {
        return new ArrayMap<>();
    }
}

File: baselibrary-kotlin/src/main/java/com/minicart/android/baselibrary/dagger2/module/AppModule.kt
package com.minicart.android.baselibrary.dagger2.module

import android.app.Application
import android.support.v4.util.ArrayMap
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(private val mApplication: Application) {

    @Singleton
    @Provides
    fun provideApplication(): Application {
        return mApplication
    }

    @Singleton
    @Provides
    fun provideGson(application: Application): Gson {
        return GsonBuilder().create()
    }

    @Singleton
    @Provides
    fun provideExtras(): Map<String, Any> {
        return ArrayMap()
    }
}
